components/more_channels/more_channels.jsx

```diff
@@ -32,25 +35,24 @@ export default class MoreChannels extends React.Component {
     constructor(props) {
         super(props);
 
-        this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
-
         this.searchTimeoutId = 0;
 
         this.state = {
             show: true,
             search: false,
-            channels: null,
+            channels: props.channels,
             serverError: null,
         };
     }
 
     componentDidMount() {
-        ChannelStore.addChangeListener(this.onChange);
-        this.props.actions.getChannels(TeamStore.getCurrentId(), 0, CHANNELS_CHUNK_SIZE * 2);
+        this.props.actions.getChannels(this.props.teamId, 0, CHANNELS_CHUNK_SIZE * 2);
     }
 
-    componentWillUnmount() {
-        ChannelStore.removeChangeListener(this.onChange);
+    componentWillReceiveProps(nextProps) {
+        if (!areObjectsEqual(nextProps.channels, this.props.channels)) {
```

Comment:
Can we eliminate `this.state.channels` altogether now? In particular, this check here seems as expensive as just letting React run the diff algorithm.